Source/Core/Runtime/Utils/DebugUtils.cs

```diff
@@ -0,0 +1,46 @@
+using UnityEngine;
+
+namespace VRBuilder.Core.Utils
+{
+    /// <summary>
+    /// Utility class for debug tools.
+    /// </summary>
+    public static class DebugUtils
+    {
+        /// <summary>
+        /// Draws a custom wire cylinder gizmo.
+        /// </summary>
+        public static void DrawWireCylinderGizmo(Vector3 startPoint, Vector3 endPoint, float width, Color color)
+        {
+            Gizmos.color = color;
+
+            // Calculate cylinder direction and length
+            Vector3 direction = endPoint - startPoint;
+            float length = direction.magnitude;
+            Vector3 directionNormalized = direction.normalized;
+
+            // Calculate rotation
+            Quaternion rotation = Quaternion.FromToRotation(Vector3.up, directionNormalized);
+            Vector3 midPoint = (startPoint + endPoint) / 2;
+
+            // Scale the cylinder to match the specified length and radius
+            Matrix4x4 oldMatrix = Gizmos.matrix;
+            Gizmos.matrix = Matrix4x4.TRS(midPoint, rotation, new Vector3(width, length / 2, width));
+
+            // Draw the cylinder
+            Gizmos.DrawWireMesh(GetCylinderMesh());
+
+            // Reset the Gizmos matrix
+            Gizmos.matrix = oldMatrix;
+        }
+
+        private static Mesh GetCylinderMesh()
+        {
+            // Generate or get a cylinder mesh
+            GameObject tempCylinder = GameObject.CreatePrimitive(PrimitiveType.Cylinder);
+            Mesh cylinderMesh = tempCylinder.GetComponent<MeshFilter>().sharedMesh;
```

Comment:
Have you thought of creating a global `private static Mesh cylinderMesh;` insted?
Just speculating: Creating multipel ones per frame could lead to some unneeded creation and deletion of a the Mesh also triggering the garbage collector.

```suggestion
            cylinderMesh = tempCylinder.GetComponent<MeshFilter>().sharedMesh;
```

Reply:
Done